### PR TITLE
Osbuild Manifests

### PR DESCRIPTION
**Pull request by admiyo**

### CODE REVIEW

**Comment by pypingou** [comment]
is the `test` folder the best location for these files?

I think I would have put them at the top of the project, in an `osbuild-manifests` folder (as it is here), next to the `package_list` folder as it is basically one of the artifact the SIG will produce

**Review by admiyo — COMMENTED** [review]
# ./do_osbuild.sh 
dirname: missing operand
Try 'dirname --help' for more information.
./do_osbuild.sh: line 10: osbuild-mpp: command not found

So we need a way to identify which version of os build we need.  I think that is currently coming out of a copr.

**Comment by juanje** [comment]
> # ./do_osbuild.sh
> 
> dirname: missing operand
> Try 'dirname --help' for more information.
> ./do_osbuild.sh: line 10: osbuild-mpp: command not found
> 
> So we need a way to identify which version of os build we need. I think that is currently coming out of a copr.

@admiyo  that's what @roobrien is working on right now: #141
Maybe we can wait until he finish to do it for one of the manifest and the add the rest.

@pypingou 
Probably that's not the best location, but, for now, we're putting them there as well (well, in `files/`), because `tests/ci/` is were all the building stuff lives. But my idea is to change the repository layout. It started with in one way, but it has grown and need some pruning and re-shaping.
But probably after we finish the migration to Gitlab. Unless it takes too much time.

**Comment by admiyo** [comment]
One thing to keep in mind is that the whole repo is getting restructured to include the microservices.  In the post-state, we will have to find a different place for them to live.  But it would be good to have a working set of manifests in the repo before that.  I think I'd like to keep the structure as is until then.

**Comment by admiyo** [comment]
I just tested the manifest build
./do_osbuild.sh osbuild-manifests/cs8/cs8-rpi4.mpp.json --export image
It creates a few artifacts that we will have to clean up.  The image goes in image_output/.  The mpp becomes a .json file in the same directory.  The interim pieces are put into  osbuild_store/.  When the time comes to build these from the pipeline, we will want to specify where to drop these artifacts.

**Comment by juanje** [comment]
> One thing to keep in mind is that the whole repo is getting restructured to include the microservices. In the post-state, we will have to find a different place for them to live. 

@admiyo Do you mean rome, berlin, etc?
All those microservices have their own repos. If you see what we have already at Gitlab, we have a subgroup for the microservices' repos and a repo for the pipeline (called `manifests`, but it will be actually the content of this repo):

https://gitlab.com/redhat/edge/ci-cd

**Comment by admiyo** [comment]
> > One thing to keep in mind is that the whole repo is getting restructured to include the microservices. In the post-state, we will have to find a different place for them to live.
> 
> @admiyo Do you mean rome, berlin, etc?
> All those microservices have their own repos. If you see what we have already at Gitlab, we have a subgroup for the microservices' repos and a repo for the pipeline (called `manifests`, but it will be actually the content of this repo):
> 
> https://gitlab.com/redhat/edge/ci-cd

Ah, yes.  Perfect.  I thought they were all going into one big repo.

Should I move this pull request to that repo?

**Comment by juanje** [comment]
> Should I move this pull request to that repo?

@admiyo We decided yesterday to keep working in this repo until we finish the migration of the pipeline (from GitHub Actions to OpenShift pipelines or Gitlab-ci), that way we don't have code changing in both repos and we don't merge any code that haven't been tested by the pipeline.

Keep in mind that, right now, the Gitlab repo doesn't run the pipeline yet, so no change there will be triggering the pipeline.

TL;DR you should keep working in this repo until we finish the pipeline (GitHub Actions) migration.

**Review by lmilbaum — APPROVED** [review]
Let's merge that PR because eventually the code would land in gitlab.com.  We should defer all discussions when the migration is done.
